### ISSUE TITLE
macros documentation yamls in /macros causes parsing error

### ISSUE THREAD

**Issue by matt-winkler** [dbt-labs/dbt-core#3613]
### Describe the bug
This is a Core bug that I observed while running dbt Cloud. It appears to be specific to v0.20.0 of dbt Core, as testing with several earlier version and the latest develop version does not trigger the same error. Creating a `.yaml` file in `macros/<filename>.yaml` as instructed [here](https://docs.getdbt.com/reference/macro-properties) triggers a partial parsing error when running v0.20.0 of dbt. 

### Steps To Reproduce
1. Set a dbt Cloud development environment to v0.20.0
2. Add the following macro in `macros/`
```
{% macro grant_all_on_schemas(schemas, role) %}
  {% for schema in schemas %}
    grant usage on schema {{ schema }} to role {{ role }};
    grant select on all tables in schema {{ schema }} to role {{ role }};
    grant select on all views in schema {{ schema }} to role {{ role }};
    grant select on future tables in schema {{ schema }} to role {{ role }};
    grant select on future views in schema {{ schema }} to role {{ role }};
  {% endfor %}
{% endmacro %}
```

3. Add a file `macros/macros_docs.yml` and this yaml to it: 
```
version: 2

macros:
  - name: grant_all_on_schemas
    description: testing docs functionality in macros path
```

### Expected behavior
dbt Cloud should compile the project without error

### Screenshots and log output
![image](https://user-images.githubusercontent.com/75497565/126813945-83e77eaf-7748-4b5a-89e1-0b475fac7579.png)

### System information
**Which database are you using dbt with?**
- [ ] postgres
- [ ] redshift
- [ ] bigquery
- [ X] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
v0.20.0
```

**Comment by jtcohen6** [comment]
@matt-winkler I haven't been able to reproduce this locally. In the dbt Cloud IDE, I did get this error:

<img width="761" alt="Screen Shot 2021-07-25 at 2 34 33 PM" src="https://user-images.githubusercontent.com/13897643/126909645-0e056c1a-c04e-4dae-bfc9-81ae77f7208f.png">

Is that the error you remember seeing? If so, we've got a bug report open for that in #3567, which we're going to resolve in #3618.

If you remember seeing a different error, could you try reproducing again, and include the full debug-level ("Details") logs?

**Comment by matt-winkler** [comment]
Hey @jtcohen6 that's the same error message, but I'm only seeing that when I have a `.yml` file with `version: 2` specified in the /macros directory with v0.20.0. Another `schema.yml` file in /models isn't tripping the error.

**Comment by jtcohen6** [comment]
@matt-winkler Weird, ok! We'll have a fix for the general issue with missing `version` in v0.20.1-rc1 (coming out this week). Let's be sure to revisit this issue once the RC is available in dbt Cloud, and confirm that it's fixed, or investigate further if not.

**Comment by gshank** [comment]
This might be the same problem as in #3627.

**Comment by gshank** [comment]
I'm pretty sure that this is fixed in 0.21 and 1.0.0, so I'm closing this ticket. Please re-open if you're still having a problem.
